mongo/mongo.go

```diff
@@ -638,10 +639,24 @@ func installPackage(pkg string, pacconfer config.PackagingConfigurer, pacman man
 
 func installMongod(operatingsystem string, numaCtl bool) error {
 	if featureflag.Enabled(feature.MongoDbSnap) {
+		snapName := ServiceName
+
+		// If we're installing a local snap, then update provide
+		// an absolute path. snap install will then do the Right Thing (TM).
+		files, err := ioutil.ReadDir("/var/lib/juju/snap")
```

Comment:
Don't use /var/lib/juju directly, should come from DataDir

Reply:
That's a good point. I've changed `installMonogo()` to accept a dataDir arg.